build/media-suite_compile.sh

```diff
@@ -1091,16 +1091,17 @@ fi
 _check=(libavif.{a,pc} avif/avif.h)
 [[ $standalone = y ]] && _check+=(bin-video/avif{enc,dec}.exe)
 if [[ $libavif = y ]] && {
-        pc_exists "aom" || pc_exists "dav1d" || pc_exists "rav1e"
+        pc_exists "aom" || pc_exists "dav1d" || pc_exists "rav1e" || pc_exists "libgav1"
     } &&
     do_vcs "https://github.com/AOMediaCodec/libavif.git"; then
     do_uninstall "${_check[@]}"
     do_pacman_install libjpeg-turbo
     do_patch "https://raw.githubusercontent.com/m-ab-s/mabs-patches/master/libavif/0001-CMake-Use-the-import-libraries-and-the-proper-variab.patch" am
     extracommands=()
+    pc_exists "libgav1" && extracommands+=("-DAVIF_CODEC_LIBGAV1=ON")
     pc_exists "dav1d" && extracommands+=("-DAVIF_CODEC_DAV1D=ON")
     pc_exists "rav1e" && extracommands+=("-DAVIF_CODEC_RAV1E=ON")
-    pc_exists "aom" && extracommands+=("-DAVIF_CODEC_AOM=ON")
+    pc_exists "aom" && extracommands+=("-DAVIF_CODEC_AOM=ON -DAVIF_CODEC_AOM_DECODE=ON -DAVIF_CODEC_AOM_ENCODE=ON")
```

Comment:
What does this change?

Reply:
In GCC manually adding the AVIF_CODEC_AOM definition doesn't activate AVIF_CODEC_AOM_DECODE and AVIF_CODEC_AOM_ENCODE. Unless they are to be inactive.

Reply:
Nope, like I thought, this doesn't do shit. Remove this line's changes.

You'd only use these options if you want to **selectively** enable/disable encoding **or** decoding with AOM, if AOM is enabled. As in, if you're enabling both dav1d/gav1 (i.e. good/fast decoders) and aom, you wouldn't use aom for decoding, so there's no point including it.

Reply:
If dav1d/libgav1 is on then it will be understood that AVIF_CODEC_AOM_DECODE must be off. What do we do with AVIF_CODEC_AOM_ENCODE encoder? Also redundant function?
I am surprised that AOM decoder doesn't work with assembler and needs other decoders for amateur purposes. GCC itself is known not to be suitable.

Reply:
#error AVIF_CODEC_AOM_DECODE or AVIF_CODEC_AOM_ENCODE must be defined
```
#if defined(AVIF_CODEC_AOM)
    { AVIF_CODEC_CHOICE_AOM,
      "aom",
      avifCodecVersionAOM,
      avifCodecCreateAOM,
#if defined(AVIF_CODEC_AOM_DECODE) && defined(AVIF_CODEC_AOM_ENCODE)
      AVIF_CODEC_FLAG_CAN_DECODE | AVIF_CODEC_FLAG_CAN_ENCODE
#elif defined(AVIF_CODEC_AOM_DECODE)
      AVIF_CODEC_FLAG_CAN_DECODE
#elif defined(AVIF_CODEC_AOM_ENCODE)
      AVIF_CODEC_FLAG_CAN_ENCODE
#else
#error AVIF_CODEC_AOM_DECODE or AVIF_CODEC_AOM_ENCODE must be defined
#endif
```

Reply:
One of them must be defined and enabled _if_ AVIF_CODEC_AOM is enabled.
So like I said, there's no need to explicitely use them in the suite. If people really want to exclude just the decoder or just the encoder, they can use [Custom Patches](https://github.com/m-ab-s/media-autobuild_suite#custom-patches) to add the extra cmake options if they want.

Remove this change.